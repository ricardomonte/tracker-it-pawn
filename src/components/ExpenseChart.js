import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgressbar } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import TrackerStyle from '../styles/ExpenseMonthly.module.css';

const ExpenseChart = ({ expense }) => {
  const date = new Date();
  const currentMonth = date.getMonth();
  const strMonth = (currentMonth + 1).toString();
  const monthlyExpenses = expense.filter((element) => (
    element.month_of_expense === strMonth));
  const total = monthlyExpenses.reduce((total, element) => (total + element.amount), 0);

  const month = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July', 'Aug', 'Sept', 'October', 'Nov', 'Dec'];
  return (
    <div className={TrackerStyle.container}>
      <h3>{`Expenses in ${month[currentMonth]}`}</h3>
      <CircularProgressbar
        value={total}
        text={`${total} euros`}
        maxValue={1000}
      />
    </div>
  );
};

ExpenseChart.propTypes = {
  expense: PropTypes.instanceOf(Object).isRequired,
};

export default ExpenseChart;
