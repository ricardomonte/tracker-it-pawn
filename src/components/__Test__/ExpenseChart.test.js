import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import MockData from '../../../tools/MockData';
import ExpenseChart from '../ExpenseChart';

test('Content from ExpenseChart matches snapshot', () => {
  const content = render(<ExpenseChart expense={MockData} />);
  expect(content.container).toMatchSnapshot();
});
