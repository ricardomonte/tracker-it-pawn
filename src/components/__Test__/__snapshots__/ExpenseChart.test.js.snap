// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from ExpenseChart matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h3>
      Expenses in May
    </h3>
    <svg
      class="CircularProgressbar "
      data-test-id="CircularProgressbar"
      viewBox="0 0 100 100"
    >
      <path
        class="CircularProgressbar-trail"
        d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
        fill-opacity="0"
        stroke-width="8"
        style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
      />
      <path
        class="CircularProgressbar-path"
        d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
        fill-opacity="0"
        stroke-width="8"
        style="stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 289.02652413026095px;"
      />
      <text
        class="CircularProgressbar-text"
        x="50"
        y="50"
      >
        0 euros
      </text>
    </svg>
  </div>
</div>
`;
